src/hooks/useActiveSubscription.test.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable dot-notation */
+import React from 'react';
+import { render, act } from '@testing-library/react';
+import { enableFetchMocks } from 'jest-fetch-mock';
+import { TokenProvider } from './useCustomerToken';
+import useActiveSubscription from './useActiveSubscription';
+import type { UseActiveSubscriptionReturnType } from './useActiveSubscription';
+import * as getActiveSubscription from 'actions/getActiveSubscription';
+
+enableFetchMocks();
+fetchMock.enableMocks();
+
+const mockPricePlan = {
+  displayName: 'Test Price Plan',
+  name: 'test_price_plan',
+  period: 'quarter',
+  basePrice: 100,
+};
+
+let hookResult: UseActiveSubscriptionReturnType | null;
+
+const MockComponent = (props?: { token?: string }): JSX.Element => {
+  const result = useActiveSubscription({
+    token: props?.token,
+  });
+
+  hookResult = result;
+
+  return <></>;
+};
+
+describe('Test `useActiveSubscription` hook', () => {
+  beforeEach(() => {
+    fetchMock.resetMocks();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    hookResult = null;
+  });
+
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('Test hook being called correctly with token provided', async () => {
```

Comment:
Tip: By convention, the strings passed to `describe` and `it` are like the subject and predicate of a sentence. `describe` gets passed the thing you're testing (the subject, i.e. `"My hook"`), and `it` gets passed the behavior you expect to see (the predicate, i.e. `"makes a network request"`). When you put the two next to each other, you should get a sentence (`"My hook makes a network request"`).

In this case, your tests would say something like this:
```ts
describe('useActiveSubscription hook', () => {
  it('accepts a token from TokenContext', () => { ... })
  it('uses a token passed in directly', () => { ... })
})
```
If you do that, Jest spits out somewhat readable sentences when running your tests.

```
useActiveSubscription hook
  - is called correctly with a token from context [✅]
  - is called correctly with token passed in directly [✅]
```

Reply:
nice tip!